lib/testers/rule-tester.js

```diff
@@ -296,9 +285,11 @@ RuleTester.prototype = {
             } else {
                 code = item.code;
 
-                // Assumes everything on the item is a config except for the
-                // parameters used by this tester
-                const itemConfig = lodash.omit(item, RuleTesterParameters);
+                // Validate only config properties
+                const schemaProperties = Object.keys(configSchema.properties);
+
+                // const schemaValidator = validator(configSchema, { verbose: true });
+                const itemConfig = lodash.pick(item, schemaProperties);
```

Comment:
:+1:

On a loosely related note, should we also validate `errors` objects to make sure they only contain allowed properties (`line`, `column`, etc.)? (This might be a separate issue, since it's not directly tied to config validation.)